matrix-sdk/src/androidTest/java/org/matrix/androidsdk/data/room/RoomDisplayNameHandlerTest.java

```diff
@@ -0,0 +1,283 @@
+
+@FixMethodOrder(MethodSorters.JVM)
+public class RoomDisplayNameHandlerTest {
+
+    @Test
+    public void RoomName_getRoomDisplayName_LL_emptyRoom() {
+        RoomName_getRoomDisplayName_emptyRoom(true);
+    }
+
+    @Test
+    public void RoomName_getRoomDisplayName_noLL_emptyRoom() {
+        RoomName_getRoomDisplayName_emptyRoom(false);
+    }
+
+    private void RoomName_getRoomDisplayName_emptyRoom(boolean withLazyLoading) {
+        Context context = InstrumentationRegistry.getContext();
+        Room room = createRoom(context, withLazyLoading, 0, false);
+
+        Assert.assertEquals("Empty room", room.getRoomDisplayName(context));
+    }
+
+    @Test
+    public void RoomName_getRoomDisplayName_noLL_roomName() {
+        RoomName_getRoomDisplayName_roomName(false);
+    }
+
+    @Test
+    public void RoomName_getRoomDisplayName_LL_roomName() {
+        RoomName_getRoomDisplayName_roomName(true);
+    }
+
+    private void RoomName_getRoomDisplayName_roomName(boolean withLazyLoading) {
+        Context context = InstrumentationRegistry.getContext();
+
+        // It does not depend on the number of users
+        for (int i = 0; i < 10; i++) {
+            Room room = createRoom(context, withLazyLoading, i, false);
+
+            room.getState().aliases = new ArrayList<>();
+            room.getState().aliases.add("Alias");
+            Assert.assertEquals("Alias", room.getRoomDisplayName(context));
+
+            // Canonical alias get priority over alias
+            room.getState().setCanonicalAlias("Canonical");
+            Assert.assertEquals("Canonical", room.getRoomDisplayName(context));
+
+            // Room name get priority over alias and canonical alias
+            room.getState().name = "Room Name";
+            Assert.assertEquals("Room Name", room.getRoomDisplayName(context));
+        }
+    }
+
+    @Test
+    public void RoomName_getRoomDisplayName_noLL_user() {
+        RoomName_getRoomDisplayName_user(false);
+    }
+
+    @Test
+    public void RoomName_getRoomDisplayName_LL_user() {
+        RoomName_getRoomDisplayName_user(true);
+    }
+
+    private void RoomName_getRoomDisplayName_user(boolean withLazyLoading) {
+        Context context = InstrumentationRegistry.getContext();
+
+        Room room;
+
+        // Only me in the room
+        room = createRoom(context, withLazyLoading, 1, false);
+        Assert.assertEquals("Empty room", room.getRoomDisplayName(context));
+
+        // One other user in the room
+        room = createRoom(context, withLazyLoading, 2, false);
+        Assert.assertEquals("UserName_2", room.getRoomDisplayName(context));
+
+        // 2 other users in the room
+        room = createRoom(context, withLazyLoading, 3, false);
+        Assert.assertEquals("UserName_2 and UserName_3", room.getRoomDisplayName(context));
+
+        room = createRoom(context, withLazyLoading, 4, false);
+        Assert.assertEquals("UserName_2 and 3 others", room.getRoomDisplayName(context));
```

Comment:
@Giom, this room has 4 users including me, and I am counted in the `3 others`. For a room with 3 members, I do not appear in the title. What do you think?